app/controllers/api/v1x0/portfolio_items_controller.rb

```diff
@@ -36,6 +36,15 @@ def destroy
         head :no_content
       end
 
+      def copy
+        access_check('read', params.require(:portfolio_item_id))
+
+        svc = Catalog::CopyPortfolioItem.new(portfolio_copy_params)
```

Comment:
@lindgrenj6 I wonder if something like this would work in a before filter
```
before_action only: [:copy] do
  access_check('read', params.require(:portfolio_item_id))
end
```
Or something like
```
before_action only: [:copy] do
  pre_access_check(params.require(:portfolio_item_id))
end
```
Where `pre_access_check` would call: `access_check('read', id))`
May be more portable for later use.

Reply:
Ok, I updated it to the latter one. I think the one-off checks like that will be used more than once so hopefully it'll be re-usable.